homeassistant/components/bring/config_flow.py

```diff
@@ -0,0 +1,99 @@
+"""Config flow for Bring! integration."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from python_bring_api.bring import Bring
+from python_bring_api.exceptions import BringAuthException, BringRequestException
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_EMAIL, CONF_PASSWORD
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.selector import (
+    TextSelector,
+    TextSelectorConfig,
+    TextSelectorType,
+)
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_EMAIL): TextSelector(
+            TextSelectorConfig(
+                type=TextSelectorType.EMAIL,
+            ),
+        ),
+        vol.Required(CONF_PASSWORD): TextSelector(
+            TextSelectorConfig(
+                type=TextSelectorType.PASSWORD,
+            ),
+        ),
+    }
+)
+
+
+async def validate_input(hass: HomeAssistant, data: dict[str, Any]) -> dict[str, Any]:
+    """Validate the user input allows us to connect.
+
+    Data has the keys from STEP_USER_DATA_SCHEMA with values provided by the user.
+    """
+    email = data[CONF_EMAIL]
+    password = data[CONF_PASSWORD]
+
+    bring = Bring(email, password)
+
+    try:
+        await hass.async_add_executor_job(bring.login)
+        await hass.async_add_executor_job(bring.loadLists)
```

Comment:
Same with these executors. I think when you make this a separate function, I think you can inline this function as it would enable you to get rid of the custom exceptions raised

Reply:
I am not sure to understand how you would get rid of the custom exceptions